apps/frontend/core-dapp/src/hooks/useSpaceId.ts

```diff
@@ -0,0 +1,56 @@
+import { useCallback, useMemo } from 'react'
+import { NETWORKS } from 'prepo-constants'
+import { createFallthroughProvider } from 'prepo-stores'
+import { usePromise } from './usePromise'
+import { useRootStore } from '../context/RootStoreProvider'
+
+type SID = {
+  getName: (address: string) => Promise<{ name: string | null }>
+}
+
+type SpaceId = {
+  dotArbName?: string
+  dotEthName?: string
+}
+
+export function useSpaceId(address: string): SpaceId {
+  const {
+    web3Store: { coreProvider },
+  } = useRootStore()
+
+  const arbitrumProvider = useMemo(
+    () =>
+      coreProvider.network.chainId === NETWORKS.arbitrumOne.chainId
+        ? coreProvider
+        : createFallthroughProvider(NETWORKS.arbitrumOne),
+    [coreProvider]
+  )
+
+  const result = usePromise(
+    useCallback(async () => {
+      // @ts-expect-error - Library does not provide types
+      const { default: SID, getSidAddress } = await import('@siddomains/sidjs')
+
```

Comment:
since arbitrumProvider is not reused I reckon we could just create it within the callback?
```suggestion
  const result = usePromise(
    useCallback(async () => {
      // @ts-expect-error - Library does not provide types
      const { default: SID, getSidAddress } = await import('@siddomains/sidjs')
      const arbProvider = coreProvider.network.chainId === NETWORKS.arbitrumOne.chainId
          ? coreProvider
          : createFallthroughProvider(NETWORKS.arbitrumOne)
```

Reply:
also the pattern is more consistent to how `ethProvider` is created